src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionSecurityTokenAuthenticator.cs

```diff
@@ -0,0 +1,1366 @@
+                = soapBindingFactory.RequireConfidentiality = soapBindingFactory.RequireIntegrity = true;
+
+                soapBindingFactory.ProtectionRequirements.IncomingSignatureParts.ChannelParts.IsBodyIncluded = true;
+                soapBindingFactory.ProtectionRequirements.OutgoingSignatureParts.ChannelParts.IsBodyIncluded = true;
+
+                MessagePartSpecification bodyPart = new MessagePartSpecification(true);
+                soapBindingFactory.ProtectionRequirements.OutgoingSignatureParts.AddParts(bodyPart, this.IssueResponseAction);
+                soapBindingFactory.ProtectionRequirements.OutgoingEncryptionParts.AddParts(bodyPart, this.IssueResponseAction);
+                soapBindingFactory.ProtectionRequirements.OutgoingSignatureParts.AddParts(bodyPart, this.RenewResponseAction);
+                soapBindingFactory.ProtectionRequirements.OutgoingEncryptionParts.AddParts(bodyPart, this.RenewResponseAction);
+
+                soapBindingFactory.ProtectionRequirements.IncomingSignatureParts.AddParts(bodyPart, this.IssueAction);
+                soapBindingFactory.ProtectionRequirements.IncomingEncryptionParts.AddParts(bodyPart, this.IssueAction);
+                soapBindingFactory.ProtectionRequirements.IncomingSignatureParts.AddParts(bodyPart, this.RenewAction);
+                soapBindingFactory.ProtectionRequirements.IncomingEncryptionParts.AddParts(bodyPart, this.RenewAction);
+            }
+
+            SupportingTokenParameters renewSupportingTokenParameters = new SupportingTokenParameters();
+            SecurityContextSecurityTokenParameters sctParameters = new SecurityContextSecurityTokenParameters();
+            sctParameters.RequireDerivedKeys = this.IssuedSecurityTokenParameters.RequireDerivedKeys;
+            renewSupportingTokenParameters.Endorsing.Add(sctParameters);
+            bootstrapSecurityProtocolFactory.SecurityBindingElement.OperationSupportingTokenParameters.Add(this.RenewAction.Value, renewSupportingTokenParameters);
+            bootstrapSecurityProtocolFactory.SecurityTokenManager = new SessionRenewSecurityTokenManager(bootstrapSecurityProtocolFactory.SecurityTokenManager, this.sessionTokenAuthenticator, (SecurityTokenResolver)this.IssuedTokenCache);
+
+           //We are passing as arguments to use existing dispatcher instead of creating another forwarding dispatcher
+           // SecurityChannelListener<TChannel> securityChannelListener = new SecurityChannelListener<TChannel>(
+           //     this.bootstrapSecurityBindingElement, this.IssuerBindingContext);
+            securityServiceDispatcher.SecurityProtocolFactory = bootstrapSecurityProtocolFactory;
+            securityServiceDispatcher.SendUnsecuredFaults = true;
+            bootstrapSecurityProtocolFactory.OpenAsync(ServiceDefaults.OpenTimeout);
+            //TODO check how can we import below utility
+            //securityServiceDispatcher.SendUnsecuredFaults = !SecurityUtils.IsCompositeDuplexBinding(context);
```

Comment:
The default here will be true as most bindings won't be a composite duplex binding and we don't have any support for that yet. As bool's default to false, you need to have this line of code there setting it to true, but leave a comment saying if/when we add support for composite duplex bindings, this will need to be false if the binding is a composite duplex binding.